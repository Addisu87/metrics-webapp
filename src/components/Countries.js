import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import { useParams } from 'react-router-dom';
import { getCountries } from '../redux/countries';
import { getContinents } from '../redux/continents';

const Countries = () => {
  const dispatch = useDispatch();
  const { continent } = useParams();
  const { continents } = useSelector((state) => state.Continents) || [];
  const countryInfo = useSelector((state) => state.Countries);

  const filteredContinent = continents.filter(
    (item) => item.id === continent,
  )[0];
  const { countries } = filteredContinent || [];
  const [country, setCountry] = useState();

  useEffect(() => {
    dispatch(getContinents());
  }, []);

  return (
    <CountryWrapper>
      <h1>{continent}</h1>
      <Select name="country" onChange={(e) => setCountry(e.target.value)}>
        <option value="" key="country">
          Country...
        </option>
        {countries
          && countries.map((land) => (
            <option value={`${land}`} key={land}>
              {land}
            </option>
          ))}
      </Select>
      <Button type="button" onClick={() => dispatch(getCountries(country))}>
        Show Information
      </Button>
      {country && country === countryInfo.id ? (
        <Info>
          <li>
            <h3>{countryInfo.id}</h3>
          </li>
          <li>
            <img src={countryInfo.flag} alt="flag" width={150} />
          </li>
          <br />
          <li>
            <strong>Cases:</strong>
            {countryInfo.cases}
          </li>
          <br />
          <li>
            <strong>Active:</strong>
            {countryInfo.active}
          </li>
          <br />
          <li>
            <strong>Critical:</strong>
            {countryInfo.critical}
          </li>
          <br />
          <li>
            <strong>Recovered:</strong>
            {countryInfo.recovered}
          </li>
          <br />
          <li>
            <strong>Deaths:</strong>
            {countryInfo.deaths}
          </li>
          <br />
          <li>
            <strong>Population:</strong>
            {countryInfo.population}
          </li>
        </Info>
      ) : (
        'Choose a country'
      )}
    </CountryWrapper>
  );
};

export default Countries;

const CountryWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-self: center;
  justify-content: center;
  margin: 2rem;
  width: 90%;
`;

const Select = styled.select`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-self: center;
  padding: 0.5rem;
  margin: 2rem;
  width: 60%;
`;

const Button = styled.button`
  padding: 0.5rem;
  color: rgb(236, 76, 138);
  border: rgb(236, 76, 138) 1px solid;
  background: #000;
  border-radius: 5px;
  margin: 2rem 0;
  align-self: center;
  width: 60%;
  &:hover {
    transition: all 0.2s ease-in-out;
    font-weight: bold;
    box-shadow: 2px 2px 10px rgb(236, 76, 138);
  }
`;

const Info = styled.ul`
  display: flex;
  flex-direction: column;
  align-self: center;
  border-radius: 5px;
  justify-content: space-between;
  width: 75%;
  list-style: none;
  background: rgb(236, 76, 138);
  color: white;
  padding: 2rem;
`;
