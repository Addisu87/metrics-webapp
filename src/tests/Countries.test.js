import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Countries from '../pages/Countries';
import store from '../redux/configureStore';
import continentsReducer from '../redux/continents';
import countriesReducer, { getCountries } from '../redux/countries';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Countries />
        </Provider>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should return content from the countries API', () => {
    const countries = countriesReducer();
    expect(countries).not.toEqual(null);
  });

  it('Should return an object with the info of the country', () => {
    const country = getCountries('Ethiopia');
    expect(country.length).toBe(1);
  });
});
