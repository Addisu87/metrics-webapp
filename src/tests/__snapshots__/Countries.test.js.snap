// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  class="sc-bczRLJ jVxtWG"
>
  <h1 />
  <select
    class="sc-gsnTZi hwveBo"
    name="country"
  >
    <option
      value=""
    >
      Country...
    </option>
  </select>
  <button
    class="sc-dkzDqf bjXjiT"
    type="button"
  >
    <h4>
      Show Information
    </h4>
  </button>
  Choose a country
</div>
`;
